### DISCUSSION TITLE
Ansi sequences

### DISCUSSION THREAD

**Discussion by silverqx** [PrismJS/prism#3450] [Q&A] [answered]
Hi, I want to ask if any of the prism components is able to parse (colorize) `ansi sequences`, something like this `\033[32mGreen text\033[0m`.

I have the table like this:
```
+------+----------------------------------------------------+-------+
| Ran? | Migration                                          | Batch |
+------+----------------------------------------------------+-------+
| Yes  | 2014_10_12_000000_create_posts_table               | 1     |
| Yes  | 2014_10_12_100000_add_factor_column_to_posts_table | 1     |
| No   | 2014_10_12_200000_create_properties_table          |       |
| No   | 2014_10_12_300000_create_phones_table              |       |
+------+----------------------------------------------------+-------+
```

And want to colorize (like it's shown in the console), like this:
![tom_migrate_status](https://user-images.githubusercontent.com/86900/166806393-0426b0c0-5d60-4ad1-9b04-d96d0073edc5.jpg)

If not, can you suggest how can I workaround this? Should I write the classic prism component, it's a little unusual case 😏.

Thank you for suggestions.

**Comment by RunDevelopment** [comment, marked as answer]
We don't have any components highlighting Ansi sequences, no. Seems like [you'll have to add that yourself](https://prismjs.com/extending.html#creating-a-new-language-definition).

That being said, if your text is literally this (no escaped Ansi sequences):

```
+------+----------------------------------------------------+-------+
| Ran? | Migration                                          | Batch |
+------+----------------------------------------------------+-------+
| Yes  | 2014_10_12_000000_create_posts_table               | 1     |
| Yes  | 2014_10_12_100000_add_factor_column_to_posts_table | 1     |
| No   | 2014_10_12_200000_create_properties_table          |       |
| No   | 2014_10_12_300000_create_phones_table              |       |
+------+----------------------------------------------------+-------+
```

then Prism won't be able to highlight anything. We can highlight text based on the characters that are actually present.

**Reply by silverqx** [reply]
> We don't have any components highlighting Ansi sequences

Ok, thx

> If your text is literally this (no escaped Ansi sequences):

Of course, I would add the ansi sequences to this table, I didn't want to paste the table with all that ansi sequences because that would be really messy.

So the only way is to write my own prism component, thank you for clarification.

**Comment by jcubic** [comment]
@silverqx Do you want to render ANSI escapes in the browser? I don't think that Prism is the right package to do that. If you want to render HTML from ANSI art I have a package like this [ansidec](https://www.npmjs.com/package/ansidec). But this package is old and it doesn't handle all ANSI escape codes. The code was extracted from [jQuery Terminal](https://terminal.jcubic.pl). But I've replaced my regex-based parser with a real [ANSI Parser](https://www.npmjs.com/package/node-ansiparser). If you want to create your own highlighter it's better to just use that parser. [unix_formatting.js](https://github.com/jcubic/jquery.terminal/blob/master/js/unix_formatting.js) file contains my processor, it translates ANSI Code into the intermediate language used by jQuery Terminal (it's called terminal formatting). If you don't want to write any extra code, you can copy/paste $.terminal.format function inside jQuery Terminal that converts terminal formatting into HTML and use unix_formatting file as is. It includes ANSI Parser.
